src/loader/mod.rs

```diff
@@ -409,11 +434,128 @@ pub fn load_cmdline<M: GuestMemory>(
     Ok(())
 }
 
+/// Writes the device tree to the given memory slice.
+///
+/// # Arguments
+///
+/// * `guest_mem` - A u8 slice that will be partially overwritten by the device tree blob.
+/// * `guest_addr` - The address in `guest_mem` at which to load the device tree blob.
+/// * `dtb_image` - The device tree blob.
+#[cfg(target_arch = "aarch64")]
+pub fn load_dtb<F, M: GuestMemory>(
```

Comment:
I think that this function should be residing somewhere higher in the design. The way I see it is: the vmm obtains the entry address for where the kernel should be loaded, it also knows about the dtb and it is the one who writes things to memory, to the dtb etc. I see this as a more decoupled and flexible design.

Reply:
This function is just a helper loading a dtb blob into memory. How to generate and modify a dtb should be implemented in vmm-fdt.